File: tests/fixtures/src/sample.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace MyApp.Services
{
    public enum ProcessStatus
    {
        Pending,
        Running,
        Done,
        Failed,
    }

    public interface IDataService
    {
        string Fetch(int id);
        Task<string> FetchAsync(int id);
        int Count { get; }
    }

    public class DataServiceBase
    {
        protected string BaseTag { get; set; } = "base";

        protected virtual string Format(string raw)
        {
            return raw.Trim();
        }
    }

    // partial class — second declaration lives in DataService.Ext.cs (separate file)
    public partial class DataService : DataServiceBase, IDataService
    {
        private readonly List<string> _cache;

        public DataService(List<string> cache)
        {
            _cache = cache;
        }

        public DataService(List<string> cache, int capacity)
        {
            _cache = cache;
            _cache.Capacity = capacity;
        }

        public int Count { get; set; }

        public ProcessStatus Status { get; set; } = ProcessStatus.Pending;

        [Authorize]
        public string Fetch(int id)
        {
            return Format(_cache[id]);
        }

        [Authorize]
        [AllowAnonymous]
        public string Fetch(int id, bool force)
        {
            if (force) _cache.Clear();
            return Fetch(id);
        }

        public async Task<string> FetchAsync(int id)
        {
            await Task.Delay(0);
            return Fetch(id);
        }

        public string Describe()
        {
            return $"{Count} items, status={Status}";
        }

        public class Config
        {
            public int MaxRetries { get; set; } = 3;
        }
    }
}

File: tests/fixtures/src/sample.cpp
#include <string>
#include <vector>
#include "core/config.h"

namespace Core {

enum class Status { Pending, Running, Done, Failed };

struct Config {
    int maxRetries;
    float timeout;
};

typedef unsigned int uint32;
using StringVec = std::vector<std::string>;

class Logger {
public:
    explicit Logger(int level);
    void log(const std::string& msg);
    int level() const;

private:
    int _level;
};

template<typename T>
class Cache {
public:
    explicit Cache(int capacity);
    void put(const std::string& key, const T& value);
    bool get(const std::string& key, T& out) const;
    void clear();

private:
    int _capacity;
};

class DataService : public Logger {
public:
    explicit DataService(const Config& cfg);
    std::string fetch(int id);
    std::string fetch(int id, bool force);
    int count() const;

private:
    Config _cfg;
};

std::string formatStatus(Status s);

}  // namespace Core
